astropy/io/fits/convenience.py

```diff
@@ -489,7 +514,9 @@ def table_to_hdu(table):
 
             col.null = fill_value.astype(table[col.name].dtype)
     else:
-        table_hdu = BinTableHDU.from_columns(np.array(table.filled()))
+        table_hdu = BinTableHDU.from_columns(np.array(table.filled()), header=hdr)
+
+    coord_meta = table.meta.pop('__coordinate_columns__', None)
```

Comment:
I think it might be more logically stored in the meta of individual columns, so it could be done as part of the unit-setting loop below, but am not completely sure. If not, do move this to be with the loop where it is used below.